share/availability/full/reconstruction_test.go

```diff
@@ -176,100 +255,141 @@ func TestShareAvailable_DisconnectedFullNodes(t *testing.T) {
+
+			var wg sync.WaitGroup
+			wg.Add(lightNodes)
+			for i := 0; i < len(lights1); i++ {
+				lights1[i] = light.Node(net)
+				go func(i int) {
+					defer wg.Done()
+					err := lights1[i].SharesAvailable(ctx, root)
+					if err != nil {
+						t.Log("light1 errors:", err)
+					}
+				}(i)
+
+				lights2[i] = light.Node(net)
+				go func(i int) {
+					defer wg.Done()
+					err := lights2[i].SharesAvailable(ctx, root)
+					if err != nil {
+						t.Log("light2 errors:", err)
+					}
+				}(i)
 			}
-		}(i)
-
-		lights2[i] = light.Node(net)
-		go func(i int) {
-			defer wg.Done()
-			err := lights2[i].SharesAvailable(ctx, root)
-			if err != nil {
-				t.Log("light2 errors:", err)
+
+			// create two full nodes and ensure they are disconnected
+			full1 := Node(net)
+			full2 := Node(net)
+			net.Disconnect(full1.ID(), full2.ID())
+
+			// ensure fulls and source are not connected
+			// so that fulls take data from light nodes only
+			net.Disconnect(full1.ID(), source.ID())
+			net.Disconnect(full2.ID(), source.ID())
+
+			// shape topology
+			for i := 0; i < len(lights1); i++ {
+				// ensure lights1 are only connected to source and full1
+				net.Connect(lights1[i].ID(), source.ID())
+				net.Connect(lights1[i].ID(), full1.ID())
+				net.Disconnect(lights1[i].ID(), full2.ID())
+				// ensure lights2 are only connected to source and full2
+				net.Connect(lights2[i].ID(), source.ID())
+				net.Connect(lights2[i].ID(), full2.ID())
+				net.Disconnect(lights2[i].ID(), full1.ID())
 			}
-		}(i)
-	}
 
-	// create two full nodes and ensure they are disconnected
-	full1 := Node(net)
-	full2 := Node(net)
-	net.Disconnect(full1.ID(), full2.ID())
-
-	// ensure fulls and source are not connected
-	// so that fulls take data from light nodes only
-	net.Disconnect(full1.ID(), source.ID())
-	net.Disconnect(full2.ID(), source.ID())
-
-	// shape topology
-	for i := 0; i < len(lights1); i++ {
-		// ensure lights1 are only connected to source and full1
-		net.Connect(lights1[i].ID(), source.ID())
-		net.Connect(lights1[i].ID(), full1.ID())
-		net.Disconnect(lights1[i].ID(), full2.ID())
-		// ensure lights2 are only connected to source and full2
-		net.Connect(lights2[i].ID(), source.ID())
-		net.Connect(lights2[i].ID(), full2.ID())
-		net.Disconnect(lights2[i].ID(), full1.ID())
+			// start reconstruction for fulls that should fail
+			ctxErr, cancelErr := context.WithTimeout(ctx, time.Second*5)
+			errg, errCtx := errgroup.WithContext(ctxErr)
+			errg.Go(func() error {
+				return full1.SharesAvailable(errCtx, root)
+			})
+			errg.Go(func() error {
+				return full2.SharesAvailable(errCtx, root)
+			})
+
+			// check that any of the fulls cannot reconstruct on their own
+			err := errg.Wait()
+			require.ErrorIs(t, err, share.ErrNotAvailable)
+			cancelErr()
+
+			// but after they connect
+			net.Connect(full1.ID(), full2.ID())
+
+			// with clean caches from the previous try
+			full1.ClearStorage()
+			full2.ClearStorage()
```

Comment:
Could you please add comment, why storage needs to be cleaned for reconstruction to happen?